uiowa/uiowa
# Enable aggregator and allow feeds to be placed via Layout Builder.

Resolves #2220, Resolves #2319 

### Testing
- Check out this branch and sync a v3 site.
  - Collegiate sites may have some confusion as the core aggregator block is available for placement.
- Go to Configuration -> Aggregator and add a few feeds. See #2220 for a list of feeds.
  - This feed has an image: http://blog.admissions.uiowa.edu/andre/feed/
- Run cron or import the feeds manually on the aggregator overview page.
- Place the custom combined feeds block via Layout Builder (now called RSS Feeds).
  - Verify items show in order of most recent.
  - Verify headline field works as expected with children.
  - Verify uploading a feed image displays next to items.
  - Verify that the "hide images" option will hide the feed images
  - Verify that the feeds are themed like the events component
  - Verify the no results text works as expected.
    - Requires deleting items from feeds.
- Test the aggregator pages and verify that they are lightly themed: https://default.local.drupal.uiowa.edu/aggregator, https://default.local.drupal.uiowa.edu/aggregator/sources, https://default.local.drupal.uiowa.edu/aggregator/sources/1

Do the above again as a webmaster to test permissions as that role. Verify you can't break anything.

Todo:
- [x] Investigate theming aggregator and aggregator items.
  - Can we make this look how we'd like?
- [x] Investigate a combined view of all aggregator feeds. Currently a route that lives at /aggregator
  - Can we alter the aggregator block or create a separate block to allow placement via LB?
- [x] Investigate aggregator feed block number of items starting at 2 instead of 1.

Assuming the above is possible, further todos include:
- [x] Lock down aggregator settings for webmaster to not change certain fields
  - Allowed html, import timings
- [x] Improve custom aggregator block to:
  - Allow single or all feed selection. 
    - Rename our block as part of this and remove the stock aggregator block from selection via LB.
  - [x] Add parent title field.
- [x] Theme aggregator and aggregator/sources routes. 
- [x] Update cron-related settings to match hourly default. Admins can override.
